core/services/workflows/syncer/handler.go

```diff
@@ -156,17 +156,22 @@ func NewEventHandler(
 	lggr logger.Logger,
 	workflowStore store.Store,
 	capRegistry core.CapabilitiesRegistry,
+	engineRegistry *EngineRegistry,
 	emitter custmsg.MessageEmitter,
 	ratelimiter *ratelimiter.RateLimiter,
 	workflowLimits *syncerlimiter.Limits,
 	workflowArtifacts WorkflowArtifactsStore,
 	opts ...func(*eventHandler),
 ) *eventHandler {
+	if engineRegistry == nil {
```

Comment:
IMO this code is misleading since a reasonable reader of this code will assume this means it's optional; in fact it isn't, since the code will behave incorrectly if we don't pass in a shared engine between the handler + syncer :)

I would remove this here in favour of passing it in at all callsites; same thing for the occurrence in the handler.